packages/compiler/src/core/logger/console-sink.ts

```diff
@@ -129,3 +132,44 @@ function getLineAndColumn(location: SourceLocation): RealLocation {
   }
   return result;
 }
+
+export async function trackAction<T>(asyncAction: () => Promise<T>, log: ProcessedLog): Promise<T> {
+  const isTTY = process.stdout?.isTTY && !process.env.CI;
+  let interval;
+  if (isTTY) {
+    const spinner = createSpinner();
+
+    interval = setInterval(() => {
+      process.stdout.clearLine(0);
+      process.stdout.cursorTo(0);
+      process.stdout.write(`\r${spinner()} ${log.message}`);
+    }, 200);
+  }
+
+  try {
+    return await asyncAction();
+  } finally {
+    if (interval) {
+      clearInterval(interval);
+      clearLastLine();
+      getChildLogs().forEach((message) => process.stdout.write(`${message}\n`));
```

Comment:
When its done I think it should show the checkmark here and not in the program.ts

Reply:
I moved the checkmark but changed the code because the message is different at the end